add hdf5 support for legion

eddy16112 opened the pull request.

junghans commented on CMakeLists.txt:
```diff
@@ -134,7 +134,13 @@ if(LEGION_USE_CUDA)
-
+option(LEGION_USE_HDF5 "enable HDF5 in Legion" OFF)
+if(LEGION_USE_HDF5)
+  set(LEGION_BUILD_EXAMPLES OFF)
```
Why are we turning the examples off when building with hdf5?

ipdemes commented:
@junghans : apparently some of their examples do not build with nvcc

junghans commented:
@ipdemes I see that in the block above, but why are we disabling it for **hdf5**?

ipdemes commented:
@ eddy16112 : do you know if enabling HDF5 in Legion breaks any legion examples?

eddy16112 commented:
@ipdemes None of them would compile if enable HDF5. Because examples requires libhdf5.so, libsz.so and libz.so. If they are installed in the same directory, then we can set HDF5_ROOT=installation_dir. As I am trying to use the HDF5 from flecsi-third-party, so before using "make install", the libhdf5.so built folder does not include libsz and libz, thus the examples can not built. I have no idea how to fix it now, so I disable examples.

junghans commented:
/cc @tuxfan 

You could add `-DHDF5_LIBRARIES=${ZLIB_LIBRARIES};${ZLIB_LIBRARIES};${ZLIB_LIBRARIES}"` to Legion's cmake options.

eddy16112 commented:
I removed the ` -DHDF5_ROOT=${HDF5_INSTALL_DIR}`,  and added `-HDF5_INCLUDE_DIR=${HDF5_INCLUDE_DIR}
   -DHDF5_LIBRARIES=${HDF5_LIBRARIES};${SZIP_LIBRARIES};${ZLIB_LIBRARIES}`, then here is the cmake output:
-- Found HDF5: /home/wwu12/flecsi/flecsi-third-party-cr-hdf5/build/hdf5/install/lib/libhdf5_debug.so;/usr/lib64/libhdf5.so;/usr/lib64/libsz.so;/usr/lib64/libz.so;/usr/lib64/libdl.so;/usr/lib64/libm.so (found version "1.8.12") found components:  C 

It does not find the SZIP and ZLIB comes with flecsi-third-party, instead, it finds the one installed in my usr/lib64, I have no idea how to fix it.

ipdemes commented:
Wei, did you fork from my branch that has hpx and legion updated?

eddy16112 commented:
@ipdemes No, I forked from master. I think it should be fine for now.

tuxfan commented:
@eddy16112 Please change all of your printf statements to use flog. For example, instead of

printf("H5Fopen failed: %lld\n", (long long)file_id);
do this
flog(error) << "H5Fopen failed: " << file_id << std::endl;

or for something like
printf("ID logical region size %ld\n", dim_size[0]);
do this
flog(info) << "ID logical region size " << dim_size[0] << std::endl;

cferenba commented:
> @eddy16112 Please change all of your printf statements to use flog....

@tuxfan : Was this comment meant for a different PR?  I see no print statements in Wei's changes here.

tuxfan commented:
@cferenba Perhaps I just noticed them now. I got some warnings while building, so I ran 'grep -r printf' in 'flecsi/io' and I get a bunch of hits. We should still fix this now.

cferenba commented:
This pull request is for flecsi-third-party, so fixing flecsi/io doesn't belong here.  I think you're right that it should be fixed, but in a flecsi PR.

tuxfan commented:
I removed the comment and sent it to Wei on slack...

tuxfan commented:
I'm trying to fix the HPX config/compile error by updating to the latest master...

ipdemes commented:
@tuxfan , @eddy16112 : This is a known issue: we found that, for some reason, new hpx choses wrong version of gcc in a docker container. I have reported that to Hartmut and they are working on reproducing and fixing the issue. I guess we can disable HPX build in flecsi-tpls until they fix it.
